src/explorer/model/Dependency.ts

```diff
@@ -15,6 +16,7 @@ export class Dependency extends TreeNode implements ITreeItem {
     private _version: string;
     private _scope: string;
     private _omittedStatus: IOmittedStatus;
+    public conflictMessages: IConflictMessage[] = [];
```

Comment:
I don't know why you need this. gid/aid are props. version1/version2 can be retrieved from omittedStatus.

Reply:
I use it to track children's `conflictMessages`.

Reply:
```
        /**
         * A flag expressing that this decoration should be
         * propagated to its parents.
         */
        propagate?: boolean;
```

you've already propagate the decoration from bottom up. why do you need children's conflict info?